src/include/cursor_inline.h

```diff
@@ -6,6 +6,32 @@
  * See the file LICENSE for redistribution information.
  */
 
+/*
+ * __wt_curhs_btree --
+ *     Convert a history store cursor to the underlying btree.
+ */
+static inline WT_BTREE *
+__wt_curhs_btree(WT_CURSOR *cursor)
+{
+    WT_CURSOR_HS *hs_cursor;
+    hs_cursor = (WT_CURSOR_HS *)cursor;
+
+    return CUR2BT(hs_cursor->file_cursor);
```

Comment:
Could you return `(WT_BTREE *)__wt_curhs_cbt`?

Reply:
Nope. `btree` is embedded deeper. Have a look at `CUR2BT()` macro.